src/dotnet/UI.Blazor/Services/VisualMediaViewerUI.cs

```diff
@@ -5,9 +5,9 @@ public class VisualMediaViewerUI(UIHub hub)
 {
     private ModalUI ModalUI => hub.ModalUI;
 
-    public async Task Show(string url, ChatEntry? chatEntry = null)
+    public async Task Show(string url, Conversation? conversation = null, ChatEntry? chatEntry = null)
```

Comment:
I think it would be better to pass an array of attachments instead of a ChatEntry or a Conversation.
The only issue i see now how to handle isYouTubeVideo check in Viewer. But to be honest, i don't understand when it can be true and why it uses link previews. May be on mobile? It should be checked or consulted with the team.